### PR TITLE
Ginkgo: Added Nightly endpoint loads tests

### PR DESCRIPTION
**Pull request by eloycoto**
Created a new Nightly test to measure how long takes to load a few
endpoints and apply policies to them. After load the endpoints, some
basic connectivity tests will run.

Related to the task #2029

Example output: 

```
• [MEASUREMENT]                                            
NightlyK8sEpsMeasurement                                   
/home/ecoto/.go/src/github.com/cilium/cilium/test/k8sT/Nightly.go:210         
  Nightly Policies                                         
  /home/ecoto/.go/src/github.com/cilium/cilium/test/k8sT/Nightly.go:209       
    Applying policies to 4 pods in bunch of 2              
    /home/ecoto/.go/src/github.com/cilium/cilium/test/k8sT/Nightly.go:208     
                                                           
    Ran 1 samples:                                         
    Runtime:                                               
      Fastest Time: 0.596s                                                    
      Slowest Time: 0.683s                                 
      Average Time: 0.639s ± 0.043s                        
    Endpoint Creation in seconds:                          
      Smallest: 0.596        
       Largest: 0.683        
       Average: 0.639 ± 0.043
    policy:                  
      Fastest Time: 49.118s  
      Slowest Time: 87.867s  
      Average Time: 68.492s ± 19.375s                      
    Policy Creation in seconds:                            
      Smallest: 49.118       
       Largest: 87.867       
       Average: 68.492 ± 19.375                            
    connTest:                
      Fastest Time: 3.299s   
      Slowest Time: 6.247s   
      Average Time: 4.773s ± 1.474s                        
    Connectivity test in seconds:                          
      Smallest: 3.299        
       Largest: 6.247        
       Average: 4.773 ± 1.474
------------------------------                             
SSSSSSSSSSSS                 
Ran 1 of 158 Specs in 210.151 seconds                      
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 157 Skipped PASS                
```

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

### CODE REVIEW

File: test/helpers/policygen/models.go [hunk 1]
```diff
@@ -677,3 +678,45 @@ func (t *TestSpec) ExecTest() error {
 	}
 	return nil
 }
+
+// TestSpecsGroup is a group of different TestSpec
+type TestSpecsGroup []*TestSpec
+
+// CreateAndApplyManifests creates all the pods manifests and applies those
```
Comment (ianvernon) [inline test/helpers/policygen/models.go]: all of the pod manifests

File: test/helpers/policygen/models.go [hunk 2]
```diff
@@ -677,3 +678,45 @@ func (t *TestSpec) ExecTest() error {
...
+// manifest to the given kubernetes instance.
```
Comment (ianvernon) [inline test/helpers/policygen/models.go]: those manifests to the given Kubernetes instance.

File: test/helpers/policygen/models.go [hunk 3]
```diff
@@ -677,3 +678,45 @@ func (t *TestSpec) ExecTest() error {
...
+	}
+	res := kub.Exec(fmt.Sprintf("cat %s > %s", strings.Join(manifests, " "), completeManifest))
+	res.ExpectSuccess()
+
+	res = kub.Exec(fmt.Sprintf("%s apply -f %s", helpers.KubectlCmd, completeManifest))
+	res.ExpectSuccess()
+}
+
+// CreateAndApplyCNP creates all Cilium Network Policies and it applies those
+// policies to the given kunernetes instance.
```
Comment (ianvernon) [inline test/helpers/policygen/models.go]: Kubernetes

File: test/k8sT/Nightly.go [hunk 4]
```diff
@@ -158,4 +159,60 @@ var _ = Describe("NightlyK8sEpsMeasurement", func() {
...
 	})
+
+	Context("Nightly Policies", func() {
+		numPods := 20
+		bunchPods := 5
+		podsCreated := 0
+
+		AfterEach(func() {
+			if CurrentGinkgoTestDescription().Failed {
+				ciliumPod, _ := kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, "k8s1")
```
Comment (ianvernon) [inline test/k8sT/Nightly.go]: Will this hardcoding of k8s1 work on the Jenkins slaves?

File: test/k8sT/Nightly.go [hunk 5]
```diff
@@ -158,4 +159,60 @@ var _ = Describe("NightlyK8sEpsMeasurement", func() {
...
+				kubectl.CiliumReport("kube-system", ciliumPod, []string{
```
Comment (ianvernon) [inline test/k8sT/Nightly.go]: Use `KubeSystemNamespace` from `cons.go`.

File: test/k8sT/Nightly.go [hunk 6]
```diff
@@ -158,4 +159,60 @@ var _ = Describe("NightlyK8sEpsMeasurement", func() {
...
+					"cilium service list",
+					"cilium endpoint list",
+					"cilium policy get"})
+			}
+
+			kubectl.Exec(fmt.Sprintf(
+				"%s delete --all pods,svc,cnp -n%s", helpers.KubectlCmd, helpers.DefaultNamespace))
+		})
+
+		Measure(fmt.Sprintf("Applying policies to %d pods in bunch of %d", numPods, bunchPods), func(b Benchmarker) {
```
Comment (ianvernon) [inline test/k8sT/Nightly.go]: What exactly does "bunch" mean?

File: test/k8sT/Nightly.go [hunk 7]
```diff
@@ -158,4 +159,60 @@ var _ = Describe("NightlyK8sEpsMeasurement", func() {
...
+			if CurrentGinkgoTestDescription().Failed {
+				ciliumPod, _ := kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, "k8s1")
+				kubectl.CiliumReport("kube-system", ciliumPod, []string{
+					"cilium service list",
+					"cilium endpoint list",
+					"cilium policy get"})
+			}
+
+			kubectl.Exec(fmt.Sprintf(
+				"%s delete --all pods,svc,cnp -n%s", helpers.KubectlCmd, helpers.DefaultNamespace))
```
Comment (ianvernon) [inline test/k8sT/Nightly.go]: `-n%s` --> `-n %s`

File: test/k8sT/Nightly.go [hunk 8]
```diff
@@ -158,4 +159,60 @@ var _ = Describe("NightlyK8sEpsMeasurement", func() {
...
+		})
+
+		Measure(fmt.Sprintf("Applying policies to %d pods in bunch of %d", numPods, bunchPods), func(b Benchmarker) {
+			testDef := func() {
+				logger.Errorf("Create %d new pods, total created are %d", numPods, podsCreated)
```
Comment (ianvernon) [inline test/k8sT/Nightly.go]: Create --> Creating

**Review by ianvernon — CHANGES_REQUESTED** [review]
Just have minor nitpicks / some questions. Before merging this, I think #2424 needs to be addressed, as we need to stabilize the existing tests before the addition of new ones.

**Review by ianvernon — APPROVED** [review]
Sorry for lagging on this, I totally missed it. LGTM!

**Comment by ianvernon** [comment]
test-me-please

**Comment by eloycoto** [comment]
@nebril  could you have a look on this please?

File: test/helpers/policygen/models.go [hunk 9]
```diff
@@ -679,3 +680,45 @@ func (t *TestSpec) ExecTest() error {
...
 	return nil
 }
+
+// TestSpecsGroup is a group of different TestSpec
+type TestSpecsGroup []*TestSpec
+
+// CreateAndApplyManifests creates all of the pods manifests and applies those
+// manifest to the given kubernetes instance.
+func (tg TestSpecsGroup) CreateAndApplyManifests(kub *helpers.Kubectl) {
+	completeManifest := "/tmp/data.yaml"
```
Comment (aanm) [inline test/helpers/policygen/models.go:692]: We should use a temporary file https://golang.org/pkg/io/ioutil/#TempFile

**Review by aanm — CHANGES_REQUESTED** [review]

**Comment by tgraf** [comment]
@eloycoto @nebril This has been laying around for a while. I would love to merge this to start collecting data on this in nightly tests. Can we move this forward?

File: test/helpers/policygen/models.go (hunk 9 above)
Comment (eloycoto) [reply test/helpers/policygen/models.go:692]: Hi @aanm the issue with the `TempFile` is that it creates the file. In this case the file will be into the VM and we can't use `TempFile` due the execution happens on the Jenkins box. 

Regards

**Comment by eloycoto** [comment]
test-all-ginkgo

**Comment by ianvernon** [comment]
test-all-ginkgo

**Comment by ianvernon** [comment]
test-nightly

**Comment by eloycoto** [comment]
test-all-ginkgo

**Comment by eloycoto** [comment]
test-nightly

**Comment by eloycoto** [comment]
Hi @tgraf 

This PR can be merged. The issues in the test are not related with the test and it's covered by task #2416 

Regards

**Pull request merged**